Zend/zend_types.h

```diff
@@ -1497,7 +1497,10 @@ static zend_always_inline uint32_t zval_delref_p(zval* pz) {
  * the Z_EXTRA space when copying property default values etc. We define separate
  * macros for this purpose, so this workaround is easier to remove in the future. */
 #define IS_PROP_UNINIT (1<<0)
-#define IS_PROP_REINITABLE (1<<1)  /* It has impact only on readonly properties */
+/* Unlocks readonly properties during cloning for exactly one assignment. It has no impact on regular properties. */
+#define IS_PROP_REINITABLE (1<<1)
+/* Re-locks readonly properties during "clone with" operations. It has no impact on regular properties. */
+#define IS_PROP_REINITED (1<<2)
```

Comment:
Unfortunately, the distinction is very subtle, but I didn't manage to come up with an easier to understand solution. Basically, the difference between them is the following:
- `IS_PROP_REINITABLE` indicates that a property is modifiable by any operation because `__clone()` unlocked the possibility. Then - after `__clone()` finishes, the possibility is gone and the flag is removed.
- `IS_PROP_REINITED` indicates that a property is not modifiable anymore in the `ZEND_CLONE_INIT_PROP` opcode. Here, we cannot rely on the absense of `IS_PROP_REINITABLE` because we cannot make these properties `IS_PROP_REINITABLE` in advance. Therefore we can only re-lock them when the assignment happened.